minimint/src/lib.rs

```diff
@@ -145,6 +173,89 @@ impl MinimintServer {
         }
     }
 
+    // Build a `ConsensusOutcome` then use the API to validate and process missed epochs
+    pub async fn rejoin_consensus(
+        &mut self,
+        _rng: &mut (impl RngCore + CryptoRng + Clone + 'static),
+    ) {
+        // Send a `RejoinRequest` message to all peers
+        self.connections
+            .send(
+                Target::AllExcept(BTreeSet::new()),
+                EpochMessage::RejoinRequest,
+            )
+            .await;
+
+        // Advance consensus to the max(next_epoch) from max_evil + 1 peers
+        let mut msg_buffer: Vec<PeerMessage> = vec![];
+        let mut next_epochs = BTreeMap::<PeerId, u64>::new();
+
+        tracing::info!("Rejoining consensus");
+        while next_epochs.len() < self.cfg.max_faulty() + 1 {
+            match self.connections.receive().await {
+                (peer, EpochMessage::RejoinEpoch(epoch)) => {
+                    next_epochs.insert(peer, epoch);
+                }
+                msg if next_epochs.contains_key(&msg.0) => msg_buffer.push(msg),
```

Comment:
Peers could have old messages in the send buffer.

But now that I think about it, it might not be safe to drop them in case the new epoch starts before a peer responds to the `RejoinRequest`